docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/How_Tos/Extend_CMS_Interface.md

```diff
@@ -204,26 +204,443 @@ To make the actions more user-friendly you can also use alternating buttons as
+work together collaboratively so that the character count only appeared when the user
+input got within a certain range of the limit. In order to do that, we'll need to be
+sure that the `TextLengthChecker` customisation is loaded ahead of the `CharacterCounter` 
+customisation. 
+
+First let's update the character counter to show characters _remaining_, which is
+much more useful. We'll also update the API to allow a `warningBuffer` prop. This is
+the amount of characters the input can be within the `limit` before the warning shows.
+
+__my-module/js/components/CharacterCounter.js__
+```js
+import React from 'react';
+
+const CharacterCounter = (TextField) => (props) => {
+    const { warningBuffer, limit, value: { length } } = props;
+    const remainingChars = limit - length;
+    const showWarning = length + warningBuffer >= limit;
+    return (
+        <div>
+            <TextField {...props} />
+            {showWarning &&
+            	<small>Characters remaining: {remainingChars}</small>
+            }
+        </div>
+    );
+}
+
+export default CharacterCounter;
+```
+
+Now, when we apply this customisation, we need to be sure it loads _after_ the length
+checker in the middleware chain, as it relies on the prop `limit`. We can do that by specifying priority using `before` and `after`
+metadata to the customisation.
+
+For this example, we'll imagine these two enhancements come from different modules.
+
+__module-a/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import CharacterCounter from './components/CharacterCounter';
+Injector.transform(
+  'my-transformation', 
+  (update) => update('TextField', CharacterCounter),
+  { after: 'my-other-transformation' }
+);
+```
+
+__module-b/js/main.js__
+```js
+import Injector from 'lib/Injector';
+import TextLengthChecker from './components/TextLengthChecker';
+
+Injector.transform(
+  'my-other-transformation', 
+  (update) => update('TextField', TextLengthChecker),
+  { before: 'my-transformation' }
+);
+```
+
+Now, both components, coming from different modules, play together nicely, in the correct
+order.
+
+![](../../../_images/react-di-3.png)
+
+`before` and `after` also accept arrays of constraints.
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (update) => update('TextField', TextLengthChecker),
+  { before: ['my-transformation', 'some-other-transformation'] }
+);
+```
+
+#### Using the * flag
+
+If you really want to be sure your customisation gets loaded first or last, you can use 
+`*` as your `before` or `after` reference. 
+
+```js
+Injector.transform(
+  'my-transformation', 
+  (update) => update('TextField', FinalTransform),
+  { after: '*' }
+);
+```
+**Note**: This flag can only be used once per transformation.
+The following are not allowed:
+* `{ before: ['*', 'something-else'] }`
+* `{ after: '*', before: 'something-else' }`
+
+### Helpful tip: Name your higher order components
```

Comment:
It might be worth including the name property in the examples above, but saying "we'll explain that 3rd argument below" or something, and then including the explanation here.

The issue being that people will likely just copy & paste the earlier examples as boilerplate.